src/NHibernate/Mapping/ByCode/ModelMapper.cs

```diff
@@ -736,7 +736,7 @@ private System.Type GetEntityBaseType(System.Type type)
 			while (analyzingType != null && analyzingType != typeof (object))
 			{
 				analyzingType = analyzingType.BaseType;
-				if (modelInspector.IsEntity(analyzingType))
+				if (modelInspector.IsDeclaredEntity(analyzingType))
 				{
 					return analyzingType;
 				}
```

Comment:
Of course :) Just trying to work out a test that will make sure that line is hit